test/Tests/src/Network/Enso_Cloud/Cloud_Tests_Setup.enso

```diff
@@ -0,0 +1,122 @@
+    httpbin_secure_client self -> HTTP =
+        password = "test-password"
+        key_store_location = enso_project.root / ".." / ".." / "tools" / "http-test-helper" / "target" / "keystore.jks"
+        key_store = KeyStore.getInstance "JKS"
+        key_store_location.with_input_stream [File_Access.Read] input_stream->
+            input_stream.with_java_stream java_stream->
+                key_store.load java_stream password.char_vector
+
+        trust_manager_factory = TrustManagerFactory.getInstance TrustManagerFactory.getDefaultAlgorithm
+        trust_manager_factory.init key_store
+
+        ssl_context = SSLContext.getInstance "TLS"
+        ssl_context.init Nothing trust_manager_factory.getTrustManagers Nothing
+        HTTP.new.set_custom_ssl_context ssl_context
+
+    is_mock self -> Boolean = case self of
+        Cloud_Tests_Setup.Mock _ _ -> True
+        _ -> False
+
+    ## Resets the user token, to avoid cached token from other tests interfering.
+    reset_token =
+        AuthenticationProvider.setToken Nothing
+
+    ## Detects the setup based on environment settings.
+    prepare : Cloud_Tests_Setup
+    prepare =
+        real_cloud = Environment.get "ENSO_RUN_REAL_CLOUD_TEST" . is_nothing . not
+        case real_cloud of
+            True ->
+                api_url = Environment.get "ENSO_CLOUD_API_URI" . if_nothing <|
+                    Panic.throw (Illegal_State.Error "If ENSO_RUN_REAL_CLOUD_TEST is defined, ENSO_CLOUD_API_URI must be defined as well.")
+                credentials_location = Environment.get "ENSO_CLOUD_CREDENTIALS_FILE" . if_nothing <|
+                    Panic.throw (Illegal_State.Error "If ENSO_RUN_REAL_CLOUD_TEST is defined, ENSO_CLOUD_CREDENTIALS_FILE must be defined as well.")
+
+                Cloud_Tests_Setup.reset_token
+                Cloud_Tests_Setup.Cloud (URI.from api_url) (File.new credentials_location)
+
+            False ->
+                base_url = Environment.get "ENSO_HTTP_TEST_HTTPBIN_URL"
+                if base_url.is_nothing then Cloud_Tests_Setup.None else
+                    with_slash = if base_url.ends_with "/" then base_url else base_url + "/"
+                    enso_cloud_url = with_slash + "enso-cloud-mock/"
+
+                    tmp_cred_file = File.create_temporary_file "enso-test-credentials" ".txt"
+                    Cloud_Tests_Setup.test_token.write tmp_cred_file
+
+                    Cloud_Tests_Setup.reset_token
+                    Cloud_Tests_Setup.Mock (URI.from enso_cloud_url) tmp_cred_file
+
+    test_token -> Text = "TEST-ENSO-TOKEN-caffee"
+
+## PRIVATE
+   A helper method that retries the action a few times, to allow tests that may fail due to propagation delays to pass.
+   This is needed, because after creating a secret, there is a slight delay before it shows up within `list`.
+   To make tests robust, we add this retry logic.
+with_retries ~action =
```

Comment:
would this make sense in HTTP soon?
Limited retries and exponential back off?

Reply:
I'm not sure if I understand the question.

In HTTP tests or in HTTP user-facing API?